src/Pagination/Connection.hack

```diff
@@ -167,8 +170,16 @@ abstract class Connection {
         if (!C\is_empty($edges)) {
             $page_info['startCursor'] = C\firstx($edges)->getCursor();
             $page_info['endCursor'] = C\lastx($edges)->getCursor();
-            $page_info['hasNextPage'] = await $this->hasNextPage($this->decodeCursor($page_info['endCursor']));
-            $page_info['hasPreviousPage'] = await $this->hasPreviousPage($this->decodeCursor($page_info['startCursor']));
+            $page_info['hasNextPage'] = await $this->hasNextPage(
+                $this->args,
+                $this->decodeCursor($page_info['startCursor']),
+                $this->decodeCursor($page_info['endCursor']),
+            );
+            $page_info['hasPreviousPage'] = await $this->hasPreviousPage(
```

Comment:
if no start cursor was provided, can we skip calling this function since it'll do unnecessary work?